pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
Oh it's not the performance I'm concerned about, I was wondering if it even
took into account proper CA validation when connecting over https. I dont
think it has a valid set of CA pem's in its configuration.

On Sun, May 31, 2015, 01:22 Vladimír Čunát <notifications@github.com> wrote:

> In pkgs/build-support/fetchurl/default.nix
> <https://github.com/NixOS/nixpkgs/pull/8082#discussion_r31388082>:
>
> > @@ -1,4 +1,4 @@
> > -{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
> > +{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
>
> Binary cache handling is done by nix itself, not nixpkgs. There, if all
> fetches (from one cache) are done in a single connection, the overhead
> should be pretty minimal. It might be considered to only fetch *.narinfo
> files securely (tiny), but the performance difference doesn't seem worth
> the bother (encoding long streams by symmetric ciphers).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/8082/files#r31388082>.
>